wlnr/client.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"bufio"
+	"io"
+	"log"
+	"net"
+	"sync"
+)
+
+// Structure to bundle the TCP connection with its packet buffer
+type Client struct {
+	// The TCP connection to the client
+	conn net.Conn
+
+	// The id of this client when refering to him in messages to the host
+	id uint8
+
+	// Mutex to avoid two Sends at the same time
+	mutex sync.Mutex
+
+	// To read data from the network
+	// I don't really like having a reader using a connection which is also
```

Comment:
That is only due to us sending so few packets and they are all small enough to fit in one TCP packet. Do not do this, it might lead to hard to understand bugs down the line. 

It should be save to just replace `ReadFull(c.conn` through `ReadFull(c.reader` everywhere and you are golden. It is fine to always read from the `reader` and write to `conn` - that behavior is well defined.

Reply:
Thanks for the hint. I suspected that something like that is possible but wasn't able to confirm it with the go documentation.